File: CS/ODataService/Models/Customers.cs
using System;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using System.Collections.Generic;

namespace WebApplication1.Models {

    [Persistent("Customers")]
    public class Customer : XPLiteObject
    {
        public Customer(Session session) : base(session) { }
        public Customer() : base(XpoDefault.Session) { }
        public override void AfterConstruction() { base.AfterConstruction(); }

        string fCustomerID;
        [Key]
        [Size(5)]
        [Nullable(false)]
        public string CustomerID {
            get { return fCustomerID; }
            set { SetPropertyValue<string>(nameof(CustomerID), ref fCustomerID, value); }
        }

        string fCompanyName;
        [Indexed(Name = @"CompanyName")]
        [Size(40)]
        [Nullable(false)]
        public string CompanyName {
            get { return fCompanyName; }
            set { SetPropertyValue<string>(nameof(CompanyName), ref fCompanyName, value); }
        }

        [Association(@"OrdersReferencesCustomers")]
        public XPCollection<Order> Orders { get { return GetCollection<Order>(nameof(Orders)); } }
    }

}

File: VB/ODataService/Models/Customers.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports System.Collections.Generic

Namespace WebApplication1.Models

    <Persistent("Customers")> _
    Public Class Customer
        Inherits XPLiteObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(XpoDefault.Session)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub

        Private fCustomerID As String
        <Key, Size(5), Nullable(False)> _
        Public Property CustomerID() As String
            Get
                Return fCustomerID
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)(NameOf(CustomerID), fCustomerID, value)
            End Set
        End Property

        Private fCompanyName As String
        <Indexed(Name := "CompanyName"), Size(40), Nullable(False)> _
        Public Property CompanyName() As String
            Get
                Return fCompanyName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)(NameOf(CompanyName), fCompanyName, value)
            End Set
        End Property

        <Association("OrdersReferencesCustomers")> _
        Public ReadOnly Property Orders() As XPCollection(Of Order)
            Get
                Return GetCollection(Of Order)(NameOf(Orders))
            End Get
        End Property
    End Class

End Namespace
